sharding/node/backend.go

```diff
@@ -148,53 +149,50 @@ func (s *ShardEthereum) Close() {
 	close(s.stop)
 }
 
-// Register appends a service constructor function to the service registry of the
+// registerService appends a service constructor function to the service registry of the
 // sharding node.
-func (s *ShardEthereum) Register(constructor sharding.ServiceConstructor) error {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	ctx := &sharding.ServiceContext{
-		Services: make(map[reflect.Type]sharding.Service),
-	}
-
-	// Copy needed for threaded access.
-	for kind, s := range s.services {
-		ctx.Services[kind] = s
-	}
-
-	service, err := constructor(ctx)
-	if err != nil {
-		return err
-	}
-
+func (s *ShardEthereum) registerService(service sharding.Service) error {
 	kind := reflect.TypeOf(service)
 	if _, exists := s.services[kind]; exists {
 		return fmt.Errorf("service already exists: %v", kind)
 	}
 	s.services[kind] = service
-
+	s.serviceTypes = append(s.serviceTypes, kind)
 	return nil
 }
 
+// fetchService returns a reference to a service stored in the service registry.
+func (s *ShardEthereum) fetchService(service interface{}) error {
+	element := reflect.ValueOf(service).Elem()
```

Comment:
Is there any way we can assert that service is a pointer? This would resolve some of @rawfalafel's concerns about copying data. 

This does copy the pointer value, but this is desired. However, if the user passes a value then they will receive a copy of the value, right? It might make sense to be a bit defensive from our future selves or other contributors.